crates/solvers/src/domain/liquidity/concentrated.rs

```diff
@@ -37,3 +37,13 @@ pub struct Amount(pub U256);
 /// [Uniswap V3 documentation](https://docs.uniswap.org/concepts/protocol/concentrated-liquidity#ticks).
 #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub struct Tick(pub i32);
+
+/// The amount of liquidity added (or, if negative, removed) when the tick is
+/// crossed going left to right.
+#[derive(Debug, Copy, Clone)]
+pub struct LiquidityNet(pub i128);
+
+/// Amount of fees accrued when using using this pool.
+/// Can be only one of these values: 5, 30 or 100 bps
+#[derive(Clone, Debug)]
+pub struct Fee(pub eth::Rational);
```

Comment:
Should this be an enum if what the comment says is true?

Reply:
The comment is actually not entirely true. (Didn't update after finding out). These are the current fees but more could theoretically be added by a DAO decision. Will update the comment.

But even if it was true I'm a bit unsure about the exact types of the uniswap v3 pool internals at every stage of the pipeline.
I guess in the driver it makes sense to use the most accurate representation possible because we don't use it in any computations but for example in the `solvers` it might make sense to store values that can only be `u128` as a `U256` if that makes the type more ergonomic.

Reply:
Depends on the specific case but in general I prefer more accurate types. Doing `.into()` on a u128 when a U256 is needed for computation isn't a big deal but trying to later go back to u128 for something that was mistakenly declared as U256 is hard.

In this case because the valid values for the fee can change any time, it makes sense to keep it as Rational.

Reply:
I think fees are [`uint24`s on-chain](https://github.com/Uniswap/v3-core/blob/05c10bf6d547d6121622ac51c457f93775e1df09/contracts/UniswapV3Pool.sol#L48), so a `num::Ratio<u32>` is probably the most accurate representation here.

> These are the current fees but more could theoretically be added by a DAO decision.

Exactly, they already added a 4th fee AFAIU.